Convert serf-feedstock to v1 feedstock

rfiorella opened the pull request:
This PR converts serf-feedstock to a v1 recipe and switch the conda build tool to rattler-build.
It has been automatically generated with [feedrattler v0.3.14](https://github.com/hadim/feedrattler).

Changes:
- [x] 📝 Converted `meta.yaml` to `recipe.yaml`
- [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
- [x] 🔧 Updated `conda-forge.yml` to use `rattler-build` and `pixi` (optional)
- [x] 🔢 Bumped the build number
- [x] 🐍 Applied temporary fixes for `python_min` and `python_version`
- [x] 🔄 Rerender the feedstock with conda-smithy
- [x] Ensured the license file is being packaged.

rfiorella committed:
Convert to v1 feedstock and use pixi as conda install tool

rfiorella committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.11.08.11.55.03

Other tools:
- conda-build 25.9.0
- rattler-build 0.49.0
- rattler-build-conda-compat 1.4.9

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

rfiorella committed:
MNT: Re-rendered with conda-smithy 3.53.0 and conda-forge-pinning 2025.11.11.22.08.03

Other tools:
- conda-build 25.9.0
- rattler-build 0.49.0
- rattler-build-conda-compat 1.4.9

rfiorella committed:
fix SConstruct.patch

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

I considered the following status checks when analyzing this PR:
 - **linter**: passed
 - **azure**: passed


Thus the PR was passing and merged! Have a great day!
